src/platform/common/stm32/pwm_output_dshot_shared.c

```diff
@@ -63,21 +63,26 @@ FAST_DATA_ZERO_INIT uint32_t inputStampUs;
 FAST_DATA_ZERO_INIT dshotTelemetryCycleCounters_t dshotDMAHandlerCycleCounters;
 #endif
 
-motorDmaOutput_t *getMotorDmaOutput(uint8_t index)
+motorDmaOutput_t *getMotorDmaOutput(unsigned index)
 {
+    if (index >= ARRAYLEN(dmaMotors)) {
+        return NULL;
+    }
     return &dmaMotors[index];
 }
 
-bool pwmDshotIsMotorIdle(uint8_t index)
+bool pwmDshotIsMotorIdle(unsigned index)
 {
     const motorDmaOutput_t *motor = getMotorDmaOutput(index);
-    return motor->protocolControl.value;
+    return motor && motor->protocolControl.value != 0;
```

Comment:
This is probably inverted (and all motor test was inverted based on my suggestion): 
```suggestion
    return motor && motor->protocolControl.value == 0;
```

Is NULL motor idle or not?
pwmDshotIsMotorNotIdle will be slightly better from copre point of view, but its name is ugly